### PR TITLE
No longer assuming that the first metadata object is the descriptive metadata in fetch_image_ids request

### PR DESCRIPTION
**Pull request by cgalarza**
Creating helpers to retrieve descriptive and structural metadata objects and using those to retrieve the metadata desired. Adding tests for new methods created and adding basic request specs for fetch_image_ids.

### CODE REVIEW

File: app/models/repo.rb [hunk 1]
```diff
@@ -389,6 +389,20 @@ def set_image_data
     self.save!
   end
 
+  # Returning MetadataSource that contains structural metadata.
+  def structural_metadata
+    metadata_builder.metadata_source.find do |metadata|
```
Comment (mksndz) [inline app/models/repo.rb:394]: i'd suggest:

```
metadata_builder.metadata_source.find_by source_type: MetadataSource::STRUCTURAL_TYPES
```

Comment (cgalarza) [reply app/models/repo.rb:394]: ooooo that's way better. I'll change it, thanks!

File: app/models/repo.rb [hunk 2]
```diff
@@ -389,6 +389,20 @@ def set_image_data
...
+      MetadataSource::STRUCTURAL_TYPES.include?(metadata.source_type)
+    end
+  end
+
+  # Returning MetadataSource that contains descriptive metadata.
+  def descriptive_metadata
+    source = metadata_builder.metadata_source.find do |metadata|
```
Comment (mksndz) [inline app/models/repo.rb:401]: likewise:

```
metadata_builder.metadata_source.find_by source_type: MetadataSource::DESCRIPTIVE_TYPES
```

File: spec/requests/repos_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          'reading_direction' => 'left-to-right',
+          'image_ids' => [
+            '/ark81431p3348gn33%2FSHA256E-s4031202--d312b4a5ef5cf37391e17f6378dcf0c84a06823d4136b2ca941c8102b2335a91.tif.jpeg',
+            '/ark81431p3348gn33%2FSHA256E-s3777716--bd9ef5508f78c9bcc5bcb867c6533cea6f9bfcd37dc4e4ae398fef513c576c94.tif.jpeg'
+          ]
+        }
+      )
+    end
+  end
+end
```
Comment (mksndz) [inline spec/requests/repos_controller_spec.rb:50]: what's the deal with these specs? TDD for future refactoring? :smiley_cat:

Comment (cgalarza) [reply spec/requests/repos_controller_spec.rb:50]: I edited this request slightly so I decided to add some basic tests.

File: app/models/repo.rb (hunk 2 above)
Comment (cgalarza) [reply app/models/repo.rb:401]: will change!

**Review by mksndz — APPROVED** [review]

**Pull request merged**